### PR TITLE
+author page

### PR DESCRIPTION
**Pull request by ha16t2**
add author

### CODE REVIEW

File: assignment2/src/app/app.module.ts [hunk 1]
```diff
@@ -1,28 +1,35 @@
...
+    AuthorEditComponent,
     
     
   ],
   imports: [
     BrowserModule,
     AppRoutingModule,
+    FormsModule,
+    
+
```
Comment (tranmanhlinh) [inline assignment2/src/app/app.module.ts]: Remove empty lines.

File: assignment2/src/app/auth/auth.component.html [hunk 2]
```diff
@@ -5,17 +5,17 @@ <h3>LOGIN</h3>
         <div class="col-xs-12 col-sm-12 col-md-12 col-lg-12">
             <div class="row">
 
-                <form name="loginForm">
+                <form name="loginForm" #authForm="ngForm" (ngSubmit)="onSubmit(authForm)">
 
                     <div class="form-group">
                         <div class="col-xs-12 col-sm-12 col-md-12 form-item">
-                            <input type="text" name="username" class="form-control" placeholder="Username: " />
+                            <input type="email" name="email" id="email" class="form-control" ngModel placeholder="Email: " required email/>
```
Comment (tranmanhlinh) [inline assignment2/src/app/auth/auth.component.html]: Recommend using Reactive Form.

File: assignment2/src/app/auth/auth.component.html [hunk 3]
```diff
@@ -5,17 +5,17 @@ <h3>LOGIN</h3>
...
                         </div>
                     </div>
 
                     <div class="form-group">
                         <div class="col-xs-12 col-sm-12 col-md-12 form-item">
-                            <input type="password" name="password" class="form-control" placeholder="Password: " />
+                            <input type="password" id="password" class="form-control" ngModel name="password" placeholder="Password: " required minlength="6" />
```
Comment (tranmanhlinh) [inline assignment2/src/app/auth/auth.component.html]: Please use Form Validator.

File: assignment2/src/app/auth/auth.component.ts [hunk 4]
```diff
@@ -11,5 +12,9 @@ export class AuthComponent implements OnInit {
 
   ngOnInit() {
   }
+  onSubmit(authForm:NgForm){
+    console.log(authForm.value);
+    authForm.reset;
```
Comment (tranmanhlinh) [inline assignment2/src/app/auth/auth.component.ts]: Remove console.log.

File: assignment2/src/app/author/author-edit/author-edit.component.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+ editMode=false;
+ editedItemIndex:number;
+ editedItem:Author;
+ @ViewChild('f',{static:false}) slForm:NgForm;
+//  @ViewChild('nameInput',{static:false}) nameInputRef:ElementRef;
+//  @ViewChild('idInput',{static:false}) idInputRef:ElementRef;
+ //authorAdd=new EventEmitter<{id:number,name:string}>();
+ 
+ //@Output() authorAdd=new EventEmitter<Author>();
+  constructor(private slService:AuthorService ) { }
```
Comment (tranmanhlinh) [inline assignment2/src/app/author/author-edit/author-edit.component.ts]: Remove all unused comments.

File: assignment2/src/app/author/author-edit/author-edit.component.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        id:this.editedItem.id,
+        name:this.editedItem.name
+      })
+    });
+  }
+  onSubmit(form:NgForm){
+    const value=form.value;
+    const newAuthor=new Author(value.id,value.name);
+   // this.authorAdd.emit(newAuthor);
+   if(this.editMode){
```
Comment (tranmanhlinh) [inline assignment2/src/app/author/author-edit/author-edit.component.ts]: Remove all unused comments.

File: assignment2/src/app/author/author-edit/author-edit.component.ts [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    this.subscription.unsubscribe();
+  }
+  onDelete(){
+    this.slService.deleteAuthor(this.editedItemIndex);
+    this.onclear();
+  }
+  onclear(){
+    this.slForm.reset();
+    this.editMode=false;
+  }
```
Comment (tranmanhlinh) [inline assignment2/src/app/author/author-edit/author-edit.component.ts]: Please install a code format plugin for your editor.

File: assignment2/src/app/author/author.component.css [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    letter-spacing: 2px;
+    background: #8AC007;
+    color: rgb(97, 207, 221);
+    border: none;
+    outline: none;
+    display: table;
+    cursor: pointer;
+    margin: 15px auto 31px;
+    border-radius: 20px 20px 20px 20px
+}
```
Comment (tranmanhlinh) [inline assignment2/src/app/author/author.component.css]: border-radius: 20px should be better?

File: assignment2/src/app/book/book-list/book-item/book-item.component.ts [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+import { Book } from 'src/app/model/book.model';
+
+@Component({
+  selector: 'app-book-item',
+  templateUrl: './book-item.component.html',
+  styleUrls: ['./book-item.component.css']
+})
+export class BookItemComponent implements OnInit {
+ @Input() book:Book[];
+ @Output() bookSeclected=new EventEmitter<void>();
```
Comment (tranmanhlinh) [inline assignment2/src/app/book/book-list/book-item/book-item.component.ts]: @Input() books : Book[]; (plural form for list name)

File: assignment2/src/app/book/book-list/book-list.component.ts [hunk 10]
```diff
@@ -1,47 +1,45 @@
...
-      'The Nuclear Express',
-      'Thomas C. Reed ',
-      2010,
-      100000,
-      'This is a political history of nuclear weapons from the discovery of fission in 1938 to the nuclear train wreck that seems to loom in our future.  It is an account of where those weapons came from, how the technology surprisingly and covertly spread, and who is likely to acquire those weapons next and most importantly why.',
-      'https://images-na.ssl-images-amazon.com/images/I/81-7CAn6M3L.jpg'),
-      
+	@Output() bookWasSelected = new EventEmitter<Book>();
+	book: Book[] = [];
+	firstName = "";
```
Comment (tranmanhlinh) [inline assignment2/src/app/book/book-list/book-list.component.ts]: Same here.

File: assignment2/src/app/services/book-service.ts [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+@Injectable({
+    providedIn: 'root'
+})
+
+export class BookService {
+    /**
+     *
+     */
+    constructor() {
```
Comment (tranmanhlinh) [inline assignment2/src/app/services/book-service.ts]: What are these lines of comment used for?

File: assignment2/src/app/book/book-list/book-list.component.ts [hunk 12]
```diff
@@ -1,47 +1,45 @@
...
+		this.bookService.getBooks().subscribe((data) => {
+			this.book = data || [];
+		})
+	}
 	
+	search() {
+		console.log('searching...');
+
 		if (this.firstName !== '') {
 			this.firstName = this.firstName.toLowerCase();
```
Comment (tranmanhlinh) [inline assignment2/src/app/book/book-list/book-list.component.ts]: if (this.firstName) { ... } should be better.

File: assignment2/src/app/app.module.ts (hunk 1 above)
Comment (ha16t2) [reply assignment2/src/app/app.module.ts]: done

File: assignment2/src/app/auth/auth.component.ts (hunk 4 above)
Comment (ha16t2) [reply assignment2/src/app/auth/auth.component.ts]: done

File: assignment2/src/app/book/book-list/book-item/book-item.component.ts (hunk 9 above)
Comment (ha16t2) [reply assignment2/src/app/book/book-list/book-item/book-item.component.ts]: done

File: assignment2/src/app/services/book-service.ts (hunk 11 above)
Comment (ha16t2) [reply assignment2/src/app/services/book-service.ts]: done

File: assignment2/src/app/book/book-list/book-list.component.ts (hunk 12 above)
Comment (ha16t2) [reply assignment2/src/app/book/book-list/book-list.component.ts]: done

File: assignment2/src/app/book/book-list/book-list.component.ts (hunk 10 above)
Comment (ha16t2) [reply assignment2/src/app/book/book-list/book-list.component.ts]: done

File: assignment2/src/app/author/author.component.css (hunk 8 above)
Comment (ha16t2) [reply assignment2/src/app/author/author.component.css]: done

File: assignment2/src/app/author/author-edit/author-edit.component.ts (hunk 5 above)
Comment (ha16t2) [reply assignment2/src/app/author/author-edit/author-edit.component.ts]: done

**Pull request merged**